src/bigunit.ts

```diff
@@ -179,8 +180,10 @@ export class BigUnit {
    * @param fraction
    * @returns new BigUnit with the result value in the same precision
    */
-  public fraction(numerator: number, denominator: number): BigUnit {
-    if (isNaN(numerator) || isNaN(denominator)) {
+  public fraction(numerator: BigUnitish, denominator: BigUnitish): BigUnit {
+    const numeratorNumber = Number(numerator);
```

Comment:
NO need to do any number checking now... but you can't necessarily cast BigUnitish into number anyway